presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -296,8 +300,8 @@ private static StringStatistics toStringStatistics(HiveWriterVersion hiveWriterV
         Slice maximum;
         Slice minimum;
         if (hiveWriterVersion != ORIGINAL) {
-            maximum = stringStatistics.hasMaximum() ? Slices.wrappedBuffer(stringStatistics.getMaximumBytes().toByteArray()) : null;
-            minimum = stringStatistics.hasMinimum() ? Slices.wrappedBuffer(stringStatistics.getMinimumBytes().toByteArray()) : null;
+            maximum = stringStatistics.hasMaximum() ? maxStringTruncateAtFirstReplacementCharacter(toSlice(stringStatistics.getMaximumBytes())) : null;
+            minimum = stringStatistics.hasMinimum() ? minStringTruncateAtFirstReplacementCharacter(toSlice(stringStatistics.getMinimumBytes())) : null;
```

Comment:
I would imagine that `maxStringTruncateAtFirstReplacementCharacter` and `minStringTruncateAtFirstReplacementCharacter` should also apply to ORIGINAL.